IntegrationTests/allocation-counter-tests-framework/template/HookedFunctionsDoHook/Sources/HookedFunctions/src/hooked-functions-unix.c

```diff
@@ -85,25 +93,37 @@ static void recursive_free(void *ptr) {
     abort();
 }
 
-/* on other UNIX systems this is slightly harder. Basically we see if we already
- * have a thread local variable that is a pointer to the original libc function.
- * If yes, easy, call it. If no, we need to resolve it which we do by using
- * dlsym. There's only one slight problem: dlsym might call back into the
- * function we're just trying to resolve (it does call malloc). In that case
- * we need to emulate that function (named recursive_*). But that's all then.
+/* on other UNIX systems this is slightly harder. Basically we already have the
+ * pointer to the original libc function in our respective global (atomic)
+ * variable.  If yes, easy, call it. If no, we need to resolve it which we do by
```

Comment:
Silly grammar nit but: you removed an "if" and now this paragraph doesn't read properly.

Reply:
Thanks, I rewrote that comment.